spec/features/braintree_credit_card_checkout_spec.rb

```diff
@@ -48,14 +48,20 @@
       within_frame("braintree-hosted-field-cvv") do
         fill_in("cvv", with: "123")
       end
-
       click_button("Save and Continue")
       within("#order_details") do
         expect(page).to have_content("CONFIRM")
       end
       click_button("Place Order")
+    end
+
+    it "checks out successfully" do
       expect(page).to have_content("Your order has been processed successfully")
     end
+
+    it "sets the payment type of source to CreditCard" do
+      expect(SolidusPaypalBraintree::Source.last.payment_type).to eq('CreditCard')
```

Comment:
Feature tests should only ever test user interactions. Since there is no reliable way to ensure that the user can see that this is the case, it should not be included in a feature test. If there was a way for a user to use the site in a manner that would cause a failure, we could test that here. Otherwise this should be tested in a controller/request test.